### ISSUE TITLE
Prevent multi-class selection for image recognition task

### ISSUE THREAD

**Issue by sid-thakur** [SkalskiP/make-sense#115]
**Describe the bug**
I have set up makesense locally. In my instance, for image recognition task, the makesense labeling interface allows selection of more than one labels.  

**To Reproduce**
1. Follow the image recognition path 
2. Upload images 
3. Add some labels
4. See labeling interface

**Expected behavior**
Image recognition should only allow one label per image. 

**Screenshots**
The output CSV file contains brackets ('[]') in class label column (attached image below). The online version of makesense does not contain the brackets. 

Multiple classes allowed for image recognition 
![image](https://user-images.githubusercontent.com/5922613/93115151-845f1e00-f689-11ea-8af6-7f6e14b73417.png)

Brackets in the Labels column (second column)
![image](https://user-images.githubusercontent.com/5922613/93114847-203c5a00-f689-11ea-832f-fce0470d875d.png)

**Desktop (please complete the following information):**
 - OS: OSX
 - Browser: FireFox

**Smartphone (please complete the following information):**
 - not applicable 

**Additional context**
We would like to replicate the exact functionality in the online version of makesense.ai. Our instance has following issues:
- allows multiple labels for image recognition task
- outputs brackets in class label data in the exported CSV file

**Comment by SkalskiP** [comment]
Hello @sid-thakur! Nice to hear that you decided to use MakeSense in your project. The behavior you have described is not a bug, but a development of the feature. 

In an earlier version MakeSene allowed to assign only one label to an image. This version is still available on https://makesense.ai, because there was no official release yet. This version of code is also currently on the `master` branch.

Shortly after the introduction of image recognition labeling functionality, an issue has popped up https://github.com/SkalskiP/make-sense/issues/96, where users have asked to add multiple labeling capabilities. The possibility was added a few weeks ago and is now waiting in the `develop` branch for another release.

In summary, the master branch behaves according to what you see on https://makesense.ai, and the development branch already has a new implementation of this functionality. This state will continue until the next release, when this and other features go to the master.

**Comment by sid-thakur** [comment]
Thank you @SkalskiP we'll account for the multi-label case

**Comment by geringer** [comment]
@SkalskiP 
Seems like a bug disguised as a half baked feature request to me. 

Having multiple labels does not work with any of the existing image classification tasks as it disturbs the back-propagation on categorical cross entropy. Pytorch converts an image folder into **one**-hot embeddings. Also, how would this be read into an torchvision.datasets.ImageFolder? Would the same file be copied into two subfolders? When you go into split the dataset into train and val, how can we ensure that these double labels don't get split up and further confuse the optimizer (X -> label_1_train could be perfect, for backprop, but would upset the algorithm on X -> label_2_validation). 

Correct me if I am wrong here, but this is a detrimental feature as a mistaken double selection of a class would upset your dataset. Lots of potential downside, no clear upside to having this feature. Perhaps @campper can provide clarity, or this should be a toggle-able feature

**Comment by SkalskiP** [comment]
Hi @geringer :) I must admit I see lots of arguments against that future now... . I need to think a little before I decide to drop it. I really would not want to bring it back once again in the future ;)

**Comment by geringer** [comment]
@SkalskiP I understand 😄  

Perhaps it would be best to just be a toggle-able feature

**Comment by SkalskiP** [comment]
@geringer toggle-mechanism is easy when you do not take into account that somebody hed started labeling in multi-mode, and then decided to turn it of... What should happen in that situation? :/

**Comment by geringer** [comment]
@SkalskiP default to single class, allow when user tries to make two labels, ask if they are sure they want multiclass labels, and then proceed with multiclass labelling

**Comment by SkalskiP** [comment]
Hi, @geringer! I think more and more about this functionality. And I am becoming more and more convinced to this solution. The problem is that MakeSense is becoming more and more complicated. And you have to start thinking about making documentation.

**Comment by sid-thakur** [comment]
Hey @SkalskiP @geringer and I work together. We were wondering if we can chat with you regarding customization of Makesense for our company's needs for data labeling. Happy to discuss the engagement/compensation, etc. You can reach me at sid.thakur@blaize.com 
Thanks
